### PR TITLE
reduce PR execution time for tabular RAI dashboard changes

### PR DESCRIPTION
**Pull request by imatiach-msft**
## Description

Reduce PR execution time for tabular RAI dashboard changes by adding path filters for text and vision notebooks and e2e test gates.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: .github/workflows/CI-e2e-notebooks-vision.yml [hunk 1]
```diff
@@ -5,6 +5,9 @@ on:
     branches: [main]
   pull_request:
     branches: [main]
+    paths:
+      - "responsibleai_vision/**"
```
Comment (gaugup) [inline .github/workflows/CI-e2e-notebooks-vision.yml:9]: Do we want to add raiwidgets are well here?

https://github.com/microsoft/responsible-ai-toolbox/blob/cd2502a459925337e5020806f96c0d3d286eab91/.github/workflows/CI-notebook-text.yml#L63

**Review by romanlutz — APPROVED** [review]

File: .github/workflows/CI-notebook-text.yml [hunk 2]
```diff
@@ -5,6 +5,9 @@ on:
     branches: [main]
   pull_request:
     branches: [main]
+    paths:
```
Comment (gaugup) [inline .github/workflows/CI-notebook-text.yml:8]: Same as above.

File: .github/workflows/CI-notebook-vision.yml [hunk 3]
```diff
@@ -5,6 +5,9 @@ on:
     branches: [main]
   pull_request:
     branches: [main]
+    paths:
```
Comment (gaugup) [inline .github/workflows/CI-notebook-vision.yml:8]: Same as above

File: .github/workflows/CI-e2e-notebooks-vision.yml (hunk 1 above)
Comment (imatiach-msft) [reply .github/workflows/CI-e2e-notebooks-vision.yml:10]: done, added raiwidgets

File: .github/workflows/CI-notebook-vision.yml (hunk 3 above)
Comment (imatiach-msft) [reply .github/workflows/CI-notebook-vision.yml:8]: done, added raiwidgets

File: .github/workflows/CI-notebook-text.yml (hunk 2 above)
Comment (imatiach-msft) [reply .github/workflows/CI-notebook-text.yml:8]: done, added raiwidgets

**Review by gaugup — APPROVED** [review]

**Pull request merged**